lua/ui/game/announcement/TitleTextAnnouncement.lua

```diff
@@ -0,0 +1,87 @@
+--******************************************************************************************************
+--** Copyright (c) 2025  Willem 'Jip' Wijnia
+--**
+--** Permission is hereby granted, free of charge, to any person obtaining a copy
+--** of this software and associated documentation files (the "Software"), to deal
+--** in the Software without restriction, including without limitation the rights
+--** to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+--** copies of the Software, and to permit persons to whom the Software is
+--** furnished to do so, subject to the following conditions:
+--**
+--** The above copyright notice and this permission notice shall be included in all
+--** copies or substantial portions of the Software.
+--**
+--** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+--** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+--** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+--** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+--** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+--** OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+--** SOFTWARE.
+--******************************************************************************************************
+
+local LayoutHelpers = import("/lua/maui/layouthelpers.lua")
+local Layouter = LayoutHelpers.ReusedLayoutFor
+
+local UIUtil = import("/lua/ui/uiutil.lua")
+local TextArea = import("/lua/ui/controls/textarea.lua").TextArea
+
+local AbstractAnnouncement = import("/lua/ui/game/announcement/AbstractAnnouncement.lua").AbstractAnnouncement
+
+--- An announcement with a title and some text below it.
+---@class UITitleTextAnnouncement : UIAbstractAnnouncement
+---@field Title Text
+---@field Text TextArea
+TitleTextAnnouncement = ClassUI(AbstractAnnouncement) {
+
+    ---@param self UITitleTextAnnouncement
+    ---@param parent Control
+    ---@param text string
+    __init = function(self, parent, text, bodyText)
+        AbstractAnnouncement.__init(self, parent)
+
+        self.Title = UIUtil.CreateText(self.ContentArea, text, 22, UIUtil.titleFont)
+        self.Text = TextArea(self.ContentArea, 600, 60)
+        self.Text:SetText(bodyText)
+    end,
+
+    ---@param self UITitleTextAnnouncement
+    ---@param parent Control
+    ---@param text string
```

Comment:
```suggestion
    ---@param text UnlocalizedString # unused
    ---@param bodyText UnlocalizedString # unused
```
Rowey used to add `# unused` comments but I'm not sure if they're needed, what do you think? You can see if a variable is unused in the function code but not in the call signature preview from the extension, that's where the comment provides new information.

Reply:
I don't think it's necessary, especially in the __post_init or __init functions as you don't see those. They are called internally.